src/provider/zeebe/properties/ExecutionListenersProps.js

```diff
@@ -0,0 +1,207 @@
+
+    modeling.updateModdleProperties(element, container, { listeners });
+  };
+}
+
+function addFactory({ bpmnFactory, commandStack, element }) {
+  return function(event) {
+    event.stopPropagation();
+
+    let commands = [];
+
+    const businessObject = getRelevantBusinessObject(element);
+
+    let extensionElements = businessObject.get('extensionElements');
+
+    // (1) ensure extension elements
+    if (!extensionElements) {
+      extensionElements = createElement(
+        'bpmn:ExtensionElements',
+        { values: [] },
+        businessObject,
+        bpmnFactory
+      );
+
+      commands.push({
+        cmd: 'element.updateModdleProperties',
+        context: {
+          element,
+          moddleElement: businessObject,
+          properties: { extensionElements }
+        }
+      });
+    }
+
+    // (2) ensure zeebe:ExecutionListeners
+    let executionListeners = getExecutionListenersContainer(businessObject);
+
+    if (!executionListeners) {
+      const parent = extensionElements;
+
+      executionListeners = createElement('zeebe:ExecutionListeners', {
+        listeners: []
+      }, parent, bpmnFactory);
+
+      commands.push({
+        cmd: 'element.updateModdleProperties',
+        context: {
+          element,
+          moddleElement: extensionElements,
+          properties: {
+            values: [ ...extensionElements.get('values'), executionListeners ]
+          }
+        }
+      });
+    }
+
+    // (3) create zeebe:ExecutionListener
+    const executionListener = createElement(
+      'zeebe:ExecutionListener', getDefaultListenerProps(element), executionListeners, bpmnFactory
+    );
+
+    // (4) add executionListener to list
+    commands.push({
+      cmd: 'element.updateModdleProperties',
+      context: {
+        element,
+        moddleElement: executionListeners,
+        properties: {
+          listeners: [ ...executionListeners.get('listeners'), executionListener ]
+        }
+      }
+    });
+
+    // (5) commit all updates
+    commandStack.execute('properties-panel.multi-command-executor', commands);
+  };
+}
+
+
+// helper //////////////////
+
+function getRelevantBusinessObject(element) {
+  let businessObject = getBusinessObject(element);
+
+  if (is(element, 'bpmn:Participant')) {
+    return businessObject.get('processRef');
+  }
+
+  return businessObject;
+}
+
+function getExecutionListenersContainer(element) {
+  const executionListeners = getExtensionElementsList(element, 'zeebe:ExecutionListeners');
+
+  return executionListeners && executionListeners[0];
+}
+
+function getListenersList(element) {
+  const executionListeners = getExecutionListenersContainer(element);
+
+  return executionListeners && executionListeners.get('listeners');
+}
+
+function canHaveExecutionListeners(bo) {
+  if (
+    isCompensationBoundaryEvent(bo) ||
+    is(bo, 'bpmn:ComplexGateway')
+  ) {
+    return false;
+  }
+
+  return isAny(bo, [
```

Comment:
Could possibly use metadata available in model: https://github.com/camunda/zeebe-bpmn-moddle/blob/main/resources/zeebe.json#L489

Reply:
I've updated the function to use the model.